go/vt/vtgate/evalengine/fn_string.go

```diff
@@ -433,3 +447,191 @@ func (call *builtinWeightString) compile(c *compiler) (ctype, error) {
 		return ctype{Type: sqltypes.VarBinary, Flag: flagNullable | flagNull, Col: collationBinary}, nil
 	}
 }
+
+func (call builtinLeftRight) eval(env *ExpressionEnv) (eval, error) {
+	str, l, err := call.arg2(env)
+	if err != nil {
+		return nil, err
+	}
+	if str == nil || l == nil {
+		return nil, nil
+	}
+
+	text, ok := str.(*evalBytes)
+	if !ok {
+		text, err = evalToVarchar(str, call.collate, true)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	length := evalToInt64(l).i
+	if length <= 0 {
+		return newEvalText(nil, text.col), nil
+	}
+
+	cs := text.col.Collation.Get().Charset()
+	strLen := charset.Length(cs, text.bytes)
```

Comment:
string length (number of runes) vs byte length. Perhaps a comment suggestion. :-D